db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLDecimal.java

```diff
@@ -537,7 +537,12 @@ protected int typeCompare(DataValueDescriptor arg) throws StandardException
     public DataValueDescriptor cloneValue(boolean forceMaterialization)
 	{
 		try {
-			return new SQLDecimal(getBigDecimal(), precision, scale);
+		    SQLDecimal decimal = new SQLDecimal(getBigDecimal(), precision, scale);
+		    if (decimal.getPrecision() != precision)
+		        decimal.setPrecision(precision);
+		    if (decimal.getScale() != scale)
+		        decimal.setScale(scale);
+			return decimal;
```

Comment:
Think the ident is spurious. Please use `{` and `}` to denote where `if` starts and ends.